sites/website/src/docs/design-systems/creating-a-component-library.md

```diff
@@ -0,0 +1,145 @@
+
+```ts
+import { html } from "@microsoft/fast-element";
+import { ElementDefinitionContext, FoundationElementDefinition } from "@microsoft/fast-foundation";
+import { Button } from "@microsoft/fast-components";
+
+// Extend the configuration with custom properties
+interface CounterDefinition extends FoundationElementDefinition {
+    defaultButtonContent?: string;
+}
+
+const counterTemplate = (context: ElementDefinitionContext, definition: CounterDefinition) => {
+  const buttonTag = context.tagFor(Button);
+
+	return html`
+    <div>The count is ${x => x.count}.</div>
+    <${buttonTag} @click=${x => x.increment()}>
+      <slot>${definition.defaultButtonContent}</slot> <!--Use the custom configuration-->
+    </${buttonTag}>
+  `;
+}
+```
+
+When the library author creates the registration function (see next section), they can provide the default content. Then, during component registration with the `DesignSystem`, an application author can optionally provide a `defaultButtonContent` field as a configuration to override this setting.
+
+### Compose and Export Registration
+
+The final step in the process is to create and export the registration function so that application authors can register the component in their `DesignSystem`. Here's how we would create the registration function for `Counter`, using our extended `CounterDefinition` configuration:
+
+```ts
+export const counter = Counter.compose<CounterDefinition>({
+  baseName: 'counter',
+  template,
+  styles,
+  defaultButtonContent: "Count!"
+});
+```
+
+Here we provide the base component name of `counter`, which will be combined with the `DesignSystem` element prefix during registration. We also provide our lazy template and styles. Finally, we specify the `defaultButtonContent` value.
+
+### Registering Library Components in an Application
+
+To register the component, an application author will import the registration and register it in their `DesignSystem`, overriding any properties as necessary:
+
+```ts
+import { counter } from "your-package";
+import { DesignSystem } from "@microsoft/fast-foundation";
+
+DesignSystem.getOrCreate()
+  .withPrefix('your')
+  .register(
+	  counter({ defaultButtonContent: "Please count." })
+  );
+```
+
+The advantage of setting `defaultButtonContent` here is that it overrides the `Counter`'s default slotted content for **all** instances of `your-counter`. Instead of having to define it everywhere in the app that the button is used, the app author provides the configuration once in a centralized location. This can be extremely useful in scenarios where components contain default icons that the app author may need to swap.
```

Comment:
is it also how some things expected to be localized/translated? if yes, good to add a note here too

Reply:
I think this *could* be used for some some localization solutions but not all, and I want to be careful about not prescribing that solution here. Good thought but I'm going to leave this as-is.